miio/device.py

```diff
@@ -238,13 +242,32 @@ def status(self) -> DeviceStatus:
         """Return device status."""
         raise NotImplementedError()
 
+    def cached_status(self) -> DeviceStatus:
```

Comment:
Let's not add caching on this PR as it isn't directly relevant to the improvements for the roborock vacuums, and it will require some more investigation on what is the best approach to implement the functionality you are aiming to fix with this.